sdk/storage/storage-blob/src/models.ts

```diff
@@ -325,3 +327,59 @@ export enum StorageBlobAudience {
    */
   DiskComputeOAuthScopes = "https://disk.compute.azure.com/.default",
 }
+
+/**
+ * Abstract representation of a poller, intended to expose just the minimal API that the user needs to work with.
+ */
+export declare interface PollerLike<TState extends PollOperationState<TResult>, TResult> {
```

Comment:
Would it be simpler to create a derived interface from SimplePollerLike? `export interface PollerLike<TState extends PollOperationState<TResult>, TResult> extends SimplePollerLike<TState, TResult> { cancelOperation: ... }`.

Reply:
`SimplePollerLike` is fundamentally different from `PollerLike` in how they define the type of the operation state, so we can't simply use/extend one in place of the other.

Reply:
I see. That makes sense to me, as OperationState is a weaker constraint than PollOperationState.